### PR TITLE
Client

### PR DESCRIPTION
**Pull request by tal2301**
Fix Download and Test Bug

### CODE REVIEW

File: src/file/file.controller.ts [hunk 1]
```diff
@@ -42,23 +43,31 @@ export class fileController {
...
 
-  public static async delete(fileId: string): Promise<IFile> {
+  public static async delete(fileId: string): Promise<any> {
     const currFile: IFile = await fileController.findById(fileId);
     const ret = await fileService.delete(fileId);
     if (ret) {
-      storageService.delete(currFile.path);
+      const res = await storageService.delete(currFile.path);
+      if (res.Errors) {
+        // TODO: Create file in DB if delete from storage fail?
```
Comment (Shahar-Y) [inline src/file/file.controller.ts]: why the todo?

File: src/file/file.controller.ts [hunk 2]
```diff
@@ -42,23 +43,31 @@ export class fileController {
...
   }
 
   public static async update(fileId: string, file: Partial<IFile>): Promise<IFile> {
     const oldFile: IFile = await fileController.findById(fileId);
     const currFile = await fileService.update(fileId, file);
     if (currFile) {
-      storageService.update(currFile.path, oldFile.path);
+      const res: any = await storageService.update(currFile.path, oldFile.path);
+      if (res.error) {
+        // TODO: Change DB update if Storage fails
```
Comment (Shahar-Y) [inline src/file/file.controller.ts]: todo?

File: src/file/file.responder.ts [hunk 3]
```diff
@@ -68,10 +67,15 @@ export class FileResponder {
...
   static async update(req: express.Request, res: express.Response) {
     const file: Partial<IFile> = req.body;
-    const ret = await fileController.update(req.params.id, file);
-    if (ret) {
-      return res.json({ success: true, return: ret });
-    }
-    throw new FileErrors.FileNotFoundError();
+    return res.json(await fileController.update(req.params.id, file));
+
+    // if (ret.error) {
```
Comment (Shahar-Y) [inline src/file/file.responder.ts]: comments

File: src/file/file.spec.ts [hunk 4]
```diff
@@ -12,7 +12,7 @@ const expect: Chai.ExpectStatic = chai.expect;
 chai.use(chaiAsPromised);
 
 const TOTAL_FILES: number = 3;
-const newName: string = 'changeName';
+const newName: string = 'changeName.txt';
```
Comment (Shahar-Y) [inline src/file/file.spec.ts]: what is that?

File: src/file/file.spec.ts [hunk 5]
```diff
@@ -69,27 +69,19 @@ describe(`File Logic`, () => {
...
+  describe.skip('#update', () => {
+    it('Should update a specific files name', async () => {
+      await fileController.update(testFiles[0]._id, { _id: testFiles[0]._id, fileName: newName, path: newName });
       const updatedFile: IFile = await fileController.findById(testFiles[0]._id);
       expect(updatedFile.fileName).to.be.equal(newName);
     });
   });
 
-  describe('#getByName', () => {
+  describe.skip('#getByName', () => {
```
Comment (Shahar-Y) [inline src/file/file.spec.ts]: why skip?

File: src/file/storage/storage.config.ts [hunk 6]
```diff
@@ -5,4 +5,4 @@ export const storageIP  : string = '40.113.65.101';
 export const bucketName : string = `${config.storage}`;
 export const accessKey  : string = 'sealminio';
 export const secretKey  : string = '/QixUY2oU6YY0f+LSjbVsmlFxfUqSn5LA7x26nSVUDKJcU7kn9kxNRY+wdLNDHBWfxnOdR8SO1XmQ2A5NRr2Uw==';
-export const writePath  : string = './tempStorage/fileName.jpg';
+// export const writePath  : string = './tempStorage/fileName.jpg';
```
Comment (Shahar-Y) [inline src/file/storage/storage.config.ts]: comments

File: src/file/storage/storage.service.ts [hunk 7]
```diff
@@ -42,27 +39,39 @@ export class storageService {
       },
     };
     // Copy the object to a new location
-    s3.copyObject(copyParams)
+    // copyObject has uncatchable error - NoSuchKey
```
Comment (Shahar-Y) [inline src/file/storage/storage.service.ts]: what does it mean?

File: src/file/storage/storage.service.ts [hunk 8]
```diff
@@ -42,27 +39,39 @@ export class storageService {
...
-      }).
-      send();
+
+    const url = s3.getSignedUrl('getObject', {
+      Bucket: bucketName,
+      Key: filePath,
+      Expires: 60
+    });
+    return url;
+    // const writePath = './tempStorage/' + filePath;
```
Comment (Shahar-Y) [inline src/file/storage/storage.service.ts]: comments

File: src/file/file.spec.ts (hunk 4 above)
Comment (tal2301) [reply src/file/file.spec.ts]: testing..

File: src/file/file.spec.ts (hunk 5 above)
Comment (tal2301) [reply src/file/file.spec.ts]: we talked about this..

**Pull request merged**